pallets/nft/src/lib.rs

```diff
@@ -296,15 +295,14 @@ impl<T: Config> Module<T> {
         }
 
         // Mints the NFT and adds it to the caller's portfolio
-        let new_balance = BalanceOf::get(&ticker, &caller_portfolio.did)
-            .checked_add(ONE_UNIT)
-            .ok_or(Error::<T>::BalanceOverflow)?;
         let nft_id = NextNFTId::try_mutate(&collection_id, try_next_pre::<T, _>)?;
-        BalanceOf::insert(&ticker, &caller_portfolio.did, new_balance);
         for (metadata_key, metadata_value) in nft_attributes.into_iter() {
             MetadataValue::insert((&collection_id, &nft_id), metadata_key, metadata_value);
         }
         PortfolioNFT::insert(caller_portfolio, (ticker, nft_id), true);
+        NumberOfNFTs::mutate(&ticker, &caller_portfolio.did, {
```

Comment:
if `NextNFTId::try_mutate(&collection_id, try_next_pre::<T, _>)?;` doesn't overflow, this won't either.

Reply:
Either add a comment to the code to explain why it is safe, or using saturating_add.  As much as possible we want to avoid standard -/+ ops in the runtime code.